### PR TITLE
New functionality for conditional validation errors

### PR DESCRIPTION
**Pull request by elimillera**
Per #78, Add argument to not error out if dataset validation fails.

Fairly simple, just adding a new argument and an if else statement

### CODE REVIEW

File: R/write.R [hunk 1]
```diff
@@ -8,6 +8,8 @@
 #' @param path Path where transport file will be written. File name sans will be
 #'   used as `xpt` name.
 #' @param label Dataset label. It must be<=40 characters.
+#' @param no_fail If TRUE, xpt validation will report a warning instead of an
+#'   error on dataset validation.
```
Comment (bms63) [inline R/write.R:12]: Should we say anything to user about the default being FALSE?

**Review by bms63 — CHANGES_REQUESTED** [review]
I'm a broken record today, but can we get a entry in `News.md` please! :)

**Comment by ddsjoberg** [comment]
Hello @elimillera ! Thank you for preparing this PR! Everything here looks perfect to me.

My only suggestion would be that we may not need an argument for this at all. I think to warn users it totally sufficient in this case. I don't know a situation where I need xportr to error due to sometimes minor non-adherence to all ADaM standards. But if someone does, they could always set `options(warn=2)` which treats warnings as errors.

Thanks again!

File: R/write.R (hunk 1 above)
Comment (ddsjoberg) [reply R/write.R:12]: If you do keep the argument, I would consider changing the name. Reading `no_fail=FALSE` is a double negative and just a bit more difficult to comprehend for users. Also, I think most users would prefer to see warnings compared to error, so I would make warning messages the default value. 🍁

**Comment by elimillera** [comment]
Thanks @ddsjoberg that's a really good point. I think sticking with the pattern of our other functions I would vote to keep the argument but I think changing the default to false is a better way to handle it. I think the arguments are useful if these is being used in a more automated pipeline where you want to stop execution if there are any issues like that in the output of the xpt file.

File: R/write.R (hunk 1 above)
Comment (elimillera) [reply R/write.R]: Updated the argument to 'strict_checks' and flipped the default

File: R/write.R [hunk 2]
```diff
@@ -8,6 +8,10 @@
 #' @param path Path where transport file will be written. File name sans will be
 #'   used as `xpt` name.
 #' @param label Dataset label. It must be<=40 characters.
```
Comment (bms63) [inline R/write.R:10]:
```suggestion
#' @param label Dataset label. It must be <=40 characters.
```

File: R/write.R [hunk 3]
```diff
@@ -8,6 +8,10 @@
...
+#' @param strict_checks If TRUE, xpt validation will report errors and not
+#'  the dataset. If FALSE, xpt validation will report warnings and continue
+#'  with writing. Defaults to FALSE
+#'
```
Comment (bms63) [inline R/write.R:14]:
```suggestion
#' @param strict_checks If TRUE, xpt validation will report errors and not
#'  write out the dataset. If FALSE, xpt validation will report warnings and continue
#'  with writing out the dataset. Defaults to FALSE
#'
```

File: R/write.R [hunk 4]
```diff
@@ -47,7 +52,10 @@ xportr_write <- function(.df, path, label = NULL) {
   checks <- xpt_validate(.df)
 
   if (length(checks) > 0) {
-    abort(c("The following validation failed:", checks))
+    if (!strict_checks)
+      warn(c("The following validation failed:", checks))
+    else
+      abort(c("The following validation failed:", checks))
```
Comment (bms63) [inline R/write.R:58]:
```suggestion
      warn(c("The following validation checks failed:", checks))
    else
      abort(c("The following validation checks failed:", checks))
```

**Review by bms63 — CHANGES_REQUESTED** [review]
Just a couple wordsmithing comments

**Review by bms63 — APPROVED** [review]
LGTM!

**Pull request merged**